### PR TITLE
merge bitcoin 15246, 15330, 15697, 16445, 16202, 16562, 18260, 22331, 23271, 16493, 17931: Refactor network message deserialization 

### PR DESCRIPTION
**Pull request by kittywhiskers**

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
13960 was already merged into develop via https://github.com/dashpay/dash/pull/4310

Does this PR need rebase? Or drop that commit?

**Comment by kittywhiskers** [comment]
Resolved, interactive rebase messed with the commit

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
Needs at least fixes listed below but `p2p_invalid_messages.py` keeps failing even after applying them (needs more backports).

**Comment by PastaPastaPasta** [comment]
Marking this as draft for two reasons:
1. Both this PR and #4330 have 14522, 14672, 14812. I believe #4330 should have priority in this regard since Konstantin was the one to mark the PRs as staged in the backport spreadsheet. Additionally, it is in an almost mergable state. Once it is merged, this PR will need to be rebased
2. CI is unhappy

**Comment by kittywhiskers** [comment]
I was initially planning to wait on #4330 because of Udjin's comment but cherry-picked the patch to test if it'll work, I'll be glad to drop those commits and rebase once Konstantin's PR is merged. +1

**Comment by vijaydasmp** [comment]
Hello, @kittywhiskers Hope you are well, I was currently blocked by bitcoin#14794: Add AddressSanitizer (ASan) Travis build, I saw the last activity on this PR 2 weeks old, If you are currently working on it, I am happy to wait

**Comment by kittywhiskers** [comment]
You can take it over, I did a partial of #14794 to resolve merges but if you're planning to do a full merge, that would be of greater value @vijaydasmp

**Comment by vijaydasmp** [comment]
Hello @kittywhiskers, Thanks for the quick response

File: test/functional/p2p_invalid_messages.py
Comment (PastaPastaPasta) [inline test/functional/p2p_invalid_messages.py:85]: imo, just drop 15005

File: test/functional/feature_assumevalid.py
Comment (PastaPastaPasta) [inline test/functional/feature_assumevalid.py:43]: imo let's just drop 16888 since we are missing feature_block stuff

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
LGTM

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**
